controllers/taloscontrolplane_controller.go

```diff
@@ -591,6 +599,25 @@ func (r *TalosControlPlaneReconciler) bootControlPlane(ctx context.Context, clus
 	return ctrl.Result{Requeue: true}, nil
 }
 
+func (r *TalosControlPlaneReconciler) bootstrapCluster(ctx context.Context, cluster *clusterv1.Cluster, machines []clusterv1.Machine) error {
+	c, err := r.talosconfigForMachines(ctx, machines...)
+	if err != nil {
+		return err
+	}
+
+	defer c.Close() //nolint:errcheck
+
+	for _, node := range c.GetEndpoints() {
```

Comment:
this looks like really wrong, we shouldn't bootstrap _all_ nodes ever (we should do just a single node).

and plus we should probably use internal IP as node address (to avoid trying to figure out which one works). Endpoints can be a mix of public/private, it doesn't matter, but to have a single address per node, we could have used Internal IP for that matter.

plus, to make things reproducible, we should sort nodes and pick the one which comes first (so it always bootstraps same node and we don't end up with split brain ever)